test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/reliefdetails/ReliefDetailsControllerSpec.scala

```diff
@@ -327,6 +355,60 @@ class ReliefDetailsControllerSpec
 
         }
 
+        "the lettings is reset" when {
+          "the private residents relief value changes" in {
+            val completeAnswers =
+              sample[CompleteReliefDetailsAnswers].copy(privateResidentsRelief = AmountInPence.fromPounds(5))
+            val fillingOutReturn          = sample[FillingOutReturn]
+            val singleDisposalDraftReturn = sample[SingleDisposalDraftReturn]
+            val disposalDate              = sample[DisposalDate]
+            val triageAnswers             = sample[CompleteSingleDisposalTriageAnswers]
+            val taxYear                   = sample[TaxYear]
+
+            val session = sessionWithReliefDetailsAnswers(
+              fillingOutReturn,
+              singleDisposalDraftReturn,
+              Some(completeAnswers),
+              None,
+              disposalDate,
+              triageAnswers,
+              taxYear
+            )._1
+
+            val updatedSession = sessionWithReliefDetailsAnswers(
+              fillingOutReturn,
+              singleDisposalDraftReturn,
+              Some(
+                IncompleteReliefDetailsAnswers(
+                  Some(AmountInPence.fromPounds(1)),
+                  None,
+                  completeAnswers.otherReliefs
+                )
+              ),
+              None,
+              disposalDate,
+              triageAnswers,
+              taxYear
+            )._1
+
+            inSequence {
+              mockAuthWithNoRetrievals()
+              mockGetSession(
+                session
+              )
+              val Some(FillingOutReturn(subscribedDetails, _, agentReferenceNumber, updatedDraftReturn)) =
```

Comment:
Yeah that's true, I am assuming it is already a Some where it might be a None

Reply:
It looks like I am already doing that in the "new overloaded sessionWithReliefDetailsAnswers". And then making sure in the storeDraftReturn that the values are send from the updatedSession. Am I missing something? :/

Reply:
I was talking about the `val Some(FillingOutReturn(...)) = updated.journeyStatus` being unsafe. Havking had a closer look, a safer way using what you have is just to use `val FillingOutReturn(...) = fillingOutReturn` - that is safer since there is no `Option` involved and `fillintOutReturn` is specifically set up as a `FillingOutReturn` rather than being typed as `JourneyStatus`. Being able to get a compile time error in case we get our types wrong is better than having a runtime MatchError.

Reply:
Ohhhh yes right, I think I've done it this time. So rather than, getting the fillingOutReturn from the session as Option[JourneyStatus], we can directly get the FillingOutReturn from the sessionWithReliefDetailsAnswers. Done.